### ISSUE TITLE
npmjs website not rendering GitHub markdown links -- fix needs baseURL in package.json

### ISSUE THREAD

**Issue by ghost** [npm/npm#18705]
#### I'm opening this issue because:
  - [X ] Other (_see below for feature requests_):

#### What's going wrong?
I have a monorepo setup for [my GitHub project](https://github.com/Typeverse/buildverse-sdk) that is similar to [babel](https://github.com/babel/babel) project layout.
This is for developing multiple npm packages within a single GitHub repo though allowing for the (sub)projects to be published on npm as separate packages.

#### How can the CLI team reproduce the problem?

My feeling is that the problem described below would be partially addressed by putting a `baseURL` field in `package.json` files (say as new child property of `repository`) which would indicate when an **npm package is not located in the root of the GitHub repo**.  This would allow the npm website developers to correctly calculate relative links from a package *README.md* to other markdown files located not only within its respective package folder in the repo but to markdown files located anywhere within the GitHub repo.

I'm not aware of any other fields in `package.json` that would help this purpose and would be gladly corrected!

**Demonstration of problem**

Please go to this page:

https://www.npmjs.com/package/buildverse-cli

At the bottom or the page there is a test link FAQ.md.

Clicking that link takes you to

https://github.com/Typeverse/buildverse-sdk/blob/HEAD/FAQ.md

which is incorrect because that page reads
buildverse-sdk
Frequently Asked Questions
and it should be the page which reads

buildverse-cli
Frequently Asked Questions
so it is linking to the wrong page in the GitHub repo.

To see what I exactly mean now go to this page:

https://github.com/Typeverse/buildverse-sdk/tree/master/buildverse-cli

This is the same README that you see for the "buildverse-cli" package page on the npm website.

Again, at the bottom of the page there is the same test link FAQ.md.

Clicking that link takes you to the correct page on GitHub, namely

https://github.com/Typeverse/buildverse-sdk/blob/HEAD/FAQ.md.

Note that this page also goes by this link

github.com/Typeverse/buildverse-sdk/blob/HEAD/buildverse-cli/FAQ.md

In summary, what I am saying is that the FAQ.md link on the "buildverse-cli" package page on the npm website

should take you to github.com/Typeverse/buildverse-sdk/blob/HEAD/buildverse-cli/FAQ.md

but it is not doing that and it is going to https://github.com/Typeverse/buildverse-sdk/blob/HEAD/FAQ.md instead which is the wrong page.
